src/implementation/base.jl

```diff
@@ -36,21 +36,22 @@ function tensoradd!(C::AbstractArray,
 
     # can we assume that C is mutable?
     # is there more functionality in base that we can use?
-    Atemp = tensoralloc_add(eltype(A), A, pA, conjA, true, allocator)
-    Ã = permutedims!(Atemp, A, linearize(pA))
+    Atemp = tensoralloc_add(eltype(A), A, pA, conjA, Val(true), allocator)
+    Atemp = permutedims!(Atemp, A, linearize(pA))
```

Comment:
Technically, there is no guarantee that the lhs is the same object as the one we allocated. In that case, we created a memory leak here. In practice this probably does not really happen, but this is definitely why I used different variable names in my initial change.

Reply:
I am not sure, I do think `permutedims!` does guarantee to store the result in the first argument, i.e. this should work without the `Atemp =` part.
```
  permutedims!(dest, src, perm)

  Permute the dimensions of array src and store the result in the array dest. perm is a vector specifying a permutation of length ndims(src). The preallocated array dest should have size(dest) == size(src)[perm] and is completely
  overwritten. No in-place permutation is supported and unexpected results will happen if src and dest have overlapping memory regions.
```